src/main/java/teammates/storage/search/StudentSearchDocument.java

```diff
@@ -0,0 +1,83 @@
+package teammates.storage.search;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrInputDocument;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.StringHelper;
+
+/**
+ * The {@link SearchDocument} object that defines how we store {@link SolrInputDocument} for students.
+ */
+class StudentSearchDocument extends SearchDocument {
+
+    private final StudentAttributes student;
+
+    StudentSearchDocument(StudentAttributes student) {
+        this.student = student;
+    }
+
+    @Override
+    SolrInputDocument toDocument() {
+        SolrInputDocument document = new SolrInputDocument();
+
+        CourseAttributes course = coursesDb.getCourse(student.course);
+
+        document.addField("id", student.key);
```

Comment:
While it was done like this before, do we have any reason to use key as the ID? This likely contributes to some test failures because the keys are semi-random and thus search documents from previous runs are not cleaned when refreshing the test data.

Reply:
Changed. This removes 2 more fails.